src/NuGet.Core/NuGet.Protocol/Resources/ServiceIndexResourceV3.cs

```diff
@@ -160,6 +164,12 @@ private static IDictionary<string, List<ServiceIndexEntry>> MakeLookup(JObject i
                         continue;
                     }
 
+                    // Capture if the resource is http & the source is https
+                    if (packageSource != null && uri.Scheme != Uri.UriSchemeHttps && packageSource.IsHttps)
```

Comment:
Oops, I approved because my suggestion was marked resolved but I just noticed this is the opposite of what I suggested.

```suggestion
                    if (packageSource != null && uri.Scheme == Uri.UriSchemeHttp && packageSource.IsHttps)
```